src/main/java/frc/robot/Robot.java

```diff
@@ -101,55 +103,113 @@ public void robotInit() {
+  }
+
+  /**
+   * A demo command that goes through all the steps of a shoot while moving algorithm Has print
+   * commands for any unimplemented functionality
+   *
+   * @param speeds
+   * @return A command that takes the robot through an auto aim sequence
+   */
   public Command autoAimDemo(Supplier<ChassisSpeeds> speeds) {
 
     return Commands.sequence(
-        Commands.runOnce(
-            () -> {
-              swerve.shotSpeeds = speeds.get();
-              swerve.curShotData =
-                  AutoAim.shotMap.get(
-                      swerve
-                          .getFuturePose(AutoAim.LOOKAHEAD_TIME)
-                          .minus(FieldConstants.getSpeaker())
-                          .getTranslation()
-                          .getNorm());
-              System.out.println(swerve.shotSpeeds.toString());
-            },
-            swerve),
-        Commands.deadline(
-            Commands.waitSeconds(1),
-            Commands.sequence(
-                Commands.waitSeconds(AutoAim.LOOKAHEAD_TIME - 0.4),
-                Commands.print("Spin Up Shooter")),
-            Commands.sequence(
-                Commands.waitSeconds(AutoAim.LOOKAHEAD_TIME - 0.7), Commands.print("Aim Shooter")),
-            Commands.sequence(
-                (swerve.pointTowardsTranslation(
-                    () -> swerve.shotSpeeds.vxMetersPerSecond,
-                    () -> swerve.shotSpeeds.vyMetersPerSecond))),
-            Commands.sequence(
-                Commands.waitSeconds(AutoAim.LOOKAHEAD_TIME - 0.1),
-                Commands.print("Rotate Robot"))),
-        Commands.print("Whoosh!"));
+            Commands.runOnce(
+                () -> {
+                  swerve.curShotSpeeds = speeds.get();
+                  swerve.curShotData =
+                      AutoAim.shotMap.get(
+                          swerve
+                              .getFuturePose(AutoAim.LOOKAHEAD_TIME, swerve.curShotSpeeds)
+                              .minus(FieldConstants.getSpeaker())
+                              .getTranslation()
+                              .getNorm());
+                },
+                swerve),
+            Commands.deadline(
+                    Commands.waitSeconds(AutoAim.LOOKAHEAD_TIME),
+                    Commands.sequence(
+                        Commands.waitSeconds(AutoAim.LOOKAHEAD_TIME - 0.4),
+                        Commands.print("Spin Up Shooter")),
+                    Commands.sequence(
+                        Commands.waitSeconds(AutoAim.LOOKAHEAD_TIME - 0.7),
+                        Commands.print("Aim Shooter")),
+                    Commands.sequence(
+                        (swerve.pointTowardsTranslationCmd(
+                            () -> swerve.curShotSpeeds.vxMetersPerSecond,
+                            () -> swerve.curShotSpeeds.vyMetersPerSecond))),
+                    Commands.sequence(
```

Comment:
can remove this print since it does actually rotate in sim. maybe move it to after the rotation?

Reply:
could also try and make it print the time since the command started